docs/language-guide.md

```diff
@@ -202,7 +202,7 @@ interface IMaterial
 struct MyCoolMaterial : IMaterial
 {
 	typedef DisneyBRDF B;
-	DisneyBRDF evalPattern(float3 position, float2 uv)
+	B evalPattern(float3 position, float2 uv)
```

Comment:
In principle the compiler should work with either `B` or `DisneyBRDF` here, because those two type names should be equivalent. In practice the compiler might have trouble with one or the other, when determining that something is a valid implementation of an interface requirement.

Did you try to compile this code and get an error message?

Reply:
I didn't compile it but when reading it, it seemed wrong, considering how typdef works and what the example tries to show. This is more fitting in my opinion and I also found it that way on this slide of the presentation "Slang: Language Mechanisms for Building Extensible Real-Time Shading Systems"
![image](https://user-images.githubusercontent.com/17255848/58896632-09c1f780-86ff-11e9-93a5-251177078444.png)

Reply:
I think it should work either way (and I'd consider it an implementation bug if it doesn't), since a typedef makes one name equivalent to another. If I have:

```
interface IThing { int getValue(); }

typedef int JustAnInt;
struct MyThing : IThing
{
    JustAnInt getValue(); // WORKS
}
```

the `MyThing.getValue()` method should be fine as an implementation of `IThing.getValue` because it returns a `JustAnInt` which is equivalent to `int`. The case for `B` is not really any different.

I take the point that using `B` might be more clear when trying to explain how associated types work, so I'm inclined to accept this change.